frontend/src/metabase-lib/filter.ts

```diff
@@ -655,7 +656,9 @@ function serializeDateTime(date: Date): string {
 }
 
 function deserializeDateTime(value: string): Date | null {
-  const dateTime = moment(value, [DATE_TIME_FORMAT, DATE_FORMAT], true);
+  const dateTime = UTC_OFFSET_REGEX.test(value)
```

Comment:
I'm reusing the same approach and regex we use in `parseTimestamp` but dates without timezone info are parsed in local timezone and not UTC, otherwise mantine components accepting `Date` instances won't work correctly.

Reply:
I believe this should work correctly in all cases since we never include the timezone when serializing the date back